src/ZohoClient.php

```diff
@@ -363,17 +363,39 @@ public function updateRecords($module, $xmlData, $id = null, $wfTrigger = null,
      * @param $module
      * @param $id
      * @param $content
+     * @param resource|string $content
      *
      * @return Response
      *
      * @throws ZohoCRMResponseException
      */
-    public function uploadFile($module, $id, $content)
+    public function uploadFile($module, $id, $content, $filename=null)
     {
-        $params['id'] = $id;
-        $params['content'] = $content;
+        $getParams['id'] = $id;
+        $postParams = [
+            [
+                'name' => 'content',
+                'contents' => $content,
+                'filename' => $filename,
```

Comment:
If we really want to be sure the empty filename is not having any impact, it would be better to write:

```
$innerPostParams = [
[
    'name' => 'content',
    'contents' => $content,
]];
if ($filename !== null) {
    $innerPostParams['filename'] = $filename;
}
$postParams = [ $innerPostParams ];
```

Alternatively (even better), you can also keep the code the way it is and write a unit test to check the behavior is ok.